app/src/split/bluetooth/central.c

```diff
@@ -344,9 +347,66 @@ static void split_central_process_connection(struct bt_conn *conn) {
 
     LOG_DBG("New connection params: Interval: %d, Latency: %d, PHY: %d", info.le.interval,
             info.le.latency, info.le.phy->rx_phy);
+
+    // Restart scanning if necessary.
+    start_scanning();
+}
+
+static int stop_scanning() {
+    LOG_DBG("Stopping peripheral scanning");
+    is_scanning = false;
+
+    int err = bt_le_scan_stop();
+    if (err) {
```

Comment:
```suggestion
    if (err < 0) {
```

We should be more diligent with these checks... Some Zephyr APIs in the past have been refactored across versions and return positive values for different success states.